src/sqlfluff/rules/structure/ST03.py

```diff
@@ -46,41 +44,32 @@ class Rule_ST03(BaseRule):
     name = "structure.unused_cte"
     aliases = ("L045",)
     groups = ("all", "core", "structure")
-    crawl_behaviour = SegmentSeekerCrawler({"statement"})
-
-    @classmethod
-    def _find_all_ctes(cls, query: Query) -> Iterator[Query]:
-        if query.ctes:
-            yield query
-        for query in query.ctes.values():
-            yield from cls._find_all_ctes(query)
-
-    @classmethod
-    def _visit_sources(cls, query: Query) -> None:
-        for selectable in query.selectables:
-            for source in query.crawl_sources(selectable.selectable, pop=True):
```

Comment:
Removing the reference to `.crawl_sources()` should have a performance boost. I can't remove the `crawl_sources()` method entirely yet though because it's still used in a few places.